datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxDetachNameGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * Provides a central logic point for generation of a detached SDX cluster name, as
+ * well as reversing a detached name.
+ */
+@Component
+public class SdxDetachNameGenerator {
+    private static final Pattern DETACHED_NAME_PATTERN = Pattern.compile(".*(-[0-9]+)$");
+
+    private static final String DELIMITER = "-";
+
+    public String generateDetachedClusterName(String originalName) {
+        return originalName + DELIMITER + new Date().getTime();
+    }
+
+    public String generateOriginalNameFromDetached(String detachedName) {
+        Matcher detachedNameMatcher = DETACHED_NAME_PATTERN.matcher(detachedName);
+        if (!detachedNameMatcher.matches()) {
```

Comment:
nit: it would help readability if you invert the if, and the return would be in the `if` branch and the exception throwing logic in the `else` part (so you would have and else branch)

Reply:
Done.